File: solutions/julia/luhn/luhn.jl
function luhn(cardnum)

    cardnum = filter(!isspace, cardnum)

    for c in cardnum
        if !isdigit(c)
            return false
        end
    end

    if length(cardnum) <= 1
        return false
    else
        cardnum = reverse(cardnum)

        odd = cardnum[1:2:end]
        even = cardnum[2:2:end]

        even = map(c -> Char('0' + (parse(Int, c) * 2 > 9 ? parse(Int, c) * 2 - 9 : parse(Int, c) * 2 )), even)

        if (sum([parse(Int, c) for c in odd]) + sum([parse(Int, c) for c in even])) % 10 == 0
            return true
        else
            return false
        end
    end
end

File: solutions/python/luhn/luhn.py
class Luhn:

    card_num = 0

    def __init__(self, card_num):
        self.card_num = card_num

    def valid(self):

        f = [c for c in self.card_num if not c.isspace()]

        for c in f:
            if not c.isdigit():
                return False

        if len(f) <= 1:
            return False
        else:
            odd = f[-1::-2]
            even = f[-2::-2]

            odd_ints = [int(c) for c in odd]
            even_ints = [int(c) for c in even]

            for i in range(0, len(even_ints)):
                if (2 * even_ints[i]) > 9:
                    even_ints[i] = 2 * even_ints[i] - 9
                else:
                    even_ints[i] = 2 * even_ints[i]

            return (sum(odd_ints) + sum(even_ints)) % 10 == 0
